lib/payment_gateways/brain_tree_blue_crypt.rb

```diff
@@ -10,27 +10,33 @@ def initialize(user)
         public_key: payment_gateway_options[:public_key],
         private_key: payment_gateway_options[:private_key]
       )
-      @gateway = @gateway_client.transparent_redirect
+      @gateway = @gateway_client.client_token
+      @customer_gateway = @gateway_client.customer
     end
 
-    def confirm(request)
-      gateway.confirm(request.query_string)
+    def confirm(options, nonce)
+      customer_gateway.update(buyer_reference_for_update, options.deep_merge({payment_method_nonce: nonce, credit_card: { options: {make_default: true, update_existing_token: current_token } }}))
     rescue Braintree::BraintreeError => e
-      notify_exception(e, request.query_string)
+      notify_exception(e, {buyer_reference: buyer_reference_for_update})
       false
     end
 
-    def form_url
-      gateway.url
+    def authorization(options={})
+      @gateway.generate(options)
     end
 
-    def create_customer_data(options)
+    def current_token
+      find_customer.credit_cards.first.try!(:token)
+    end
+
+    def create_customer_data(options={})
       customer = find_customer
       if customer
         remote_update_credit_card(customer, options)
       else
-        gateway.create_customer_data(options.merge(customer: { id: buyer_reference_for_update }))
+        customer = customer_gateway.create(options.merge(id: buyer_reference_for_update))
       end
+      customer
```

Comment:
Right, we don't care about what the method returns anymore. So I guess it's OK that now we always return the customer data and never more the response of `#remote_update_credit_card` in some cases. 👍